File: src/main/java/com/deploy/bemyplan/service/user/UserServiceUtils.java
package com.deploy.bemyplan.service.user;

import com.deploy.bemyplan.common.exception.ErrorCode;
import com.deploy.bemyplan.common.exception.model.ConflictException;
import com.deploy.bemyplan.domain.user.UserRepository;
import com.deploy.bemyplan.domain.user.UserSocialType;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import static com.deploy.bemyplan.common.exception.ErrorCode.CONFLICT_NICKNAME_EXCEPTION;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class UserServiceUtils {

    static void validateNotExistsUserName(UserRepository userRepository, String name) {
        if (userRepository.existsByName(name)) {
            throw new ConflictException(String.format("이미 존재하는 닉네임 (%s) 입니다", name), CONFLICT_NICKNAME_EXCEPTION);
        }
    }

    static void validateNotExistsUser(UserRepository userRepository, String socialId, UserSocialType socialType) {
        if (userRepository.existsBySocialIdAndSocialType(socialId, socialType)) {
            throw new ConflictException(String.format("이미 존재하는 유저 (%s - %s) 입니다", socialId, socialType), ErrorCode.CONFLICT_USER_EXCEPTION);
        }
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/bemyplan/auth/application/UserServiceUtils.kt
package com.bemyplan.auth.application

import com.deploy.bemyplan.common.exception.model.ConflictException
import com.deploy.bemyplan.domain.user.UserRepository
import com.deploy.bemyplan.domain.user.UserSocialType

internal fun validateNotExistsUserName(userRepository: UserRepository, name: String) {
    if (userRepository.existsByName(name)) {
        throw ConflictException("이미 존재하는 닉네임 ${name} 입니다")
    }
}

internal fun validateNotExistsUser(userRepository: UserRepository, socialId: String, socialType: UserSocialType) {
    if (userRepository.existsBySocialIdAndSocialType(socialId, socialType)) {
        throw ConflictException("이미 존재하는 유저 (${socialId} - ${socialType}) 입니다")
    }
}
